Makefile

```
@@ -27,6 +31,44 @@ test-ami-ubuntu: fmt-ami validate-ami-ubuntu build-ami-ubuntu ## fmt, validate,
 .PHONY: test-ami-rhel
 test-ami-rhel: fmt-ami validate-ami-rhel build-ami-rhel ## fmt, validate, and build the AMI for AWS.
 
+.PHONY: e2e-ubuntu
+e2e-ubuntu: validate-ami-ubuntu publish-ami-ubuntu
+	TEST_AMI_ID=$$(jq -r '.builds[-1].artifact_id' $(AWS_DIR)/manifest.json | cut -d ":" -f2); \
+	cd $(TEST_TF_DIR); \
+	terraform init -force-copy \
+		-backend-config="bucket=uds-ci-state-bucket" \
+		-backend-config="key=tfstate/ci/install/$${SHA:0:7}-packer-ubuntu-rke2.tfstate" \
+		-backend-config="region=us-west-2" \
+		-backend-config="dynamodb_table=uds-ci-state-dynamodb"; \
+	terraform apply -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	kubectl get nodes; \
+	terraform destroy -auto-approve; \
+	snapshot_ids=$$(aws ec2 describe-images --image-ids "$${TEST_AMI_ID}" | jq -r .Images[].BlockDeviceMappings[].Ebs.SnapshotId); \
+	aws ec2 deregister-image --image-id $${TEST_AMI_ID}; \
+	for snapshot in $${snapshot_ids}; do \
+		echo "$${snapshot}"; \
+		aws ec2 delete-snapshot --snapshot-id "$${snapshot}"; \
+	done
+
+.PHONY: e2e-rhel
+e2e-rhel: validate-ami-rhel publish-ami-rhel
+	TEST_AMI_ID=$$(jq -r '.builds[-1].artifact_id' $(AWS_DIR)/manifest.json | cut -d ":" -f2); \
+	cd $(TEST_TF_DIR); \
+	terraform init -force-copy \
+		-backend-config="bucket=uds-ci-state-bucket" \
+		-backend-config="key=tfstate/ci/install/$${SHA:0:7}-packer-rhel-rke2.tfstate" \
+		-backend-config="region=us-west-2" \
+		-backend-config="dynamodb_table=uds-ci-state-dynamodb"; \
+	terraform apply -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	kubectl get nodes; \
+	terraform destroy -auto-approve; \
```

```suggestion
	terraform destroy -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
```
I think maybe? Looks like the teardown failed here.